### DISCUSSION TITLE
how to list files from private supabase buckets

### DISCUSSION THREAD

**Discussion by Emil1733** [supabase/supabase#31393] [Questions]
I have an application on Flutterflow with Supabase DB. I want to list the documents from the private Supabase bucket with the policy to only show the files for auth.user only. 

My questions are:

How do you list the documents from that bucket? Is there a specific guide to it?
Can I populate the URL of uploaded files from storage to another table, so that I can filter them more easily?

**Comment by GaryAustin1** [comment]
You can use a list call:  https://supabase.com/docs/reference/dart/storage-from-list
You would need to meet select RLS policy and that can limit a user to see only their own files.

You can also query storage.objects directly (if you meet select RLS) to get files in a bucket.  You would need to write the code and use where clause to filter on bucket and owner_id.
